lib/customer-deposit-wallet/test/unit/Cardano/Wallet/Deposit/Write/KeysSpec.hs

```diff
@@ -0,0 +1,107 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# OPTIONS_GHC -Wno-incomplete-uni-patterns #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+-- |
+-- Copyright: © 2024 Cardano Foundation
+-- License: Apache-2.0
+--
+-- Property tests for the deposit wallet.
+module Cardano.Wallet.Deposit.Write.KeysSpec
+    ( spec
+    ) where
+
+import Prelude
+
+import Cardano.Crypto.Wallet
+    ( generate
+    )
+import Cardano.Wallet.Address.BIP32_Ed25519
+    ( XPrv
+    , XPub
+    , sign
+    , toXPub
+    )
+import Cardano.Wallet.Address.Encoding
+    ( EnterpriseAddr (..)
+    , NetworkTag (..)
+    , compactAddrFromEnterpriseAddr
+    , credentialFromXPub
+    )
+import Cardano.Wallet.Deposit.Write.Keys
+    ( enterpriseAddressFromVKey
+    , signedDSIGNfromXSignature
+    , vkeyFromXPub
+    )
+import Test.Hspec
+    ( Spec
+    , describe
+    , it
+    )
+import Test.QuickCheck
+    ( Arbitrary (..)
+    , Blind (..)
+    , Property
+    , property
+    , vectorOf
+    , withMaxSuccess
+    , (===)
+    )
+
+import qualified Cardano.Crypto.Hash.Blake2b as Hash
+import qualified Cardano.Crypto.Hash.Class as Hash
+import qualified Cardano.Ledger.BaseTypes as L
+import qualified Cardano.Ledger.Hashes as L
+import qualified Cardano.Ledger.Keys as L
+import qualified Cardano.Wallet.Read as Read
+import qualified Data.ByteString as BS
+
+{-----------------------------------------------------------------------------
+    Spec
+------------------------------------------------------------------------------}
+spec :: Spec
+spec = do
+    describe "commutes with ledger" $ do
+        it "address" $ lessCryptography $ property $
+            \xpub ->
+                enterpriseAddressFromVKey L.Mainnet (vkeyFromXPub xpub)
```

Comment:
only `L.Mainnet` is enough ?

Reply:
Seems trivial to cover more networks, we should probably add these generators